syncer/syncer.go

```diff
@@ -0,0 +1,141 @@
+package syncer
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/0xPolygon/polygon-edge/blockchain"
+	"github.com/0xPolygon/polygon-edge/helper/progress"
+	"github.com/0xPolygon/polygon-edge/syncer/peers"
+	"github.com/0xPolygon/polygon-edge/types"
+	"github.com/hashicorp/go-hclog"
+)
+
+const (
+	LoggerName = "syncer"
+)
+
+type Syncer interface {
+	Start()
+	GetSyncProgression() *progress.Progression
+	BestPeer() *peers.BestPeer
+	HasSyncPeer() bool
+	BulkSync(context.Context, func(*types.Block)) error
+	WatchSync(context.Context, func(*types.Block) bool) error
+}
+
+type Blockchain interface {
+	// Subscribe new block event
+	SubscribeEvents() blockchain.Subscription
+	// Get latest header
+	Header() *types.Header
+	// Verify fetched block
+	VerifyFinalizedBlock(*types.Block) error
+	// Write block to chain
+	WriteBlock(*types.Block) error
+}
+
+type Progression interface {
+	StartProgression(startingBlock uint64, subscription blockchain.Subscription)
+	UpdateHighestProgression(highestBlock uint64)
+	GetProgression() *progress.Progression
+	StopProgression()
+}
```

Comment:
Please notice that you are defining an interface that other modules will use for interacting with the syncer (`Syncer` interface in this package), but **are also defining interfaces you require from other modules such as blockchain and progression**.

Please look into keeping these interfaces where they belong 🙏 

cc @dbrajovic

Reply:
@zivkovicmilos 
I don't think this is duplicate definition. for example, we need to write mock that implements all methods defined in the interface in blockchain package when using interface of blockchain defined in blockchain package. Basically the use of blockchain depends on behaviour of the module.

Reply:
Of course, no need to define the interface if it is not subset of the original